File: server/scratch/get_sqlite_schema.js
const sqlite3 = require('sqlite3').verbose();
const path = require('path');

const dbPath = path.join(__dirname, '../database.sqlite');
const db = new sqlite3.Database(dbPath);

console.log('Reading SQLite database schema...');

db.all("SELECT sql FROM sqlite_master WHERE type='table' AND name='auditor_counts';", (err, rows) => {
  if (err) {
    console.error(err);
    return;
  }
  console.log('SQLite Schema:', rows);
});

db.all("SELECT * FROM sqlite_master WHERE type='trigger';", (err, rows) => {
  if (err) {
    console.error(err);
    return;
  }
  console.log('Triggers:', rows);
});

db.close();

File: server/scratch/get_sqlite_schema.py
import sqlite3
import os

db_path = os.path.join(os.path.dirname(__file__), '../database.sqlite')
conn = sqlite3.connect(db_path)
cursor = conn.cursor()

print("auditor_counts schema:")
cursor.execute("SELECT sql FROM sqlite_master WHERE name='auditor_counts'")
row = cursor.fetchone()
if row:
    print(row[0])
else:
    print("Not found")

print("\nTriggers:")
cursor.execute("SELECT name, sql FROM sqlite_master WHERE type='trigger'")
for r in cursor.fetchall():
    print(r[0], r[1])

conn.close()
